### PR TITLE
setup happkit config

### PR DESCRIPTION
**Pull request by alexcarpenter**
### What

Adding Happykit config setup to match developer https://github.com/hashicorp/dev-portal/pull/1185

### Why

To enable AB testing.

### Screenshots

----------

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [ ] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. tfc).
- [ ] Description links to related pull requests or issues, if any.

#### Content
- [ ] Redirects have been added for moved, renamed, or deleted pages.
- [ ] Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages.
- [ ] Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [ ] Pages with related content are updated and link to this content when appropriate.
- [ ] New pages have metadata (page name and description) at the top.
- [ ] New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.
- [ ] New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [ ] UI elements (button names, page names, etc.) are bolded.
- [ ] The Vercel website preview successfully deployed.

#### Reviews
- [ ] I or someone else reviewed the content for technical accuracy.
- [ ] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

File: flags/client.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import {
+	createUseFlags,
```
Comment (dstaley) [inline flags/client.ts:2]: Can you run prettier on these? Looks like some files have tabs and others have spaces.

File: next.config.js [hunk 2]
```diff
@@ -2,6 +2,11 @@ const withHashicorp = require('@hashicorp/platform-nextjs-plugin')
 const path = require('path')
 const redirects = require('./redirects.next.js')
 
+// Set api key for Happy Kit feature flags
+const happyKitKey = process.env.NEXT_PUBLIC_FLAGS_ENV_KEY
+  ? process.env.NEXT_PUBLIC_FLAGS_ENV_KEY
+  : 'flags_pub_development_343442393171755603'
```
Comment (dstaley) [inline next.config.js:8]: Just confirming: this is the key for devdot?

**Review by dstaley — CHANGES_REQUESTED** [review]
Besides the prettier stuff, LGTM!

File: next.config.js (hunk 2 above)
Comment (alexcarpenter) [reply next.config.js:8]: It is yes.

File: flags/client.ts (hunk 1 above)
Comment (alexcarpenter) [reply flags/client.ts]: https://github.com/hashicorp/terraform-website/pull/2421/commits/dbb6a8a872beac206f261842551bf06e4917176a

**Review by dstaley — APPROVED** [review]

**Pull request merged**